lib/phoenix/router.ex

```diff
@@ -771,7 +771,8 @@ defmodule Phoenix.Router do
       end
 
   """
-  defmacro forward(path, plug, plug_opts \\ [], router_opts \\ []) do
+  defmacro forward(path, plug, plug_opts \\ [], router_opts \\ [])
+  defmacro forward(path, {_, _, [mod]} = plug, plug_opts, router_opts) when is_atom(mod) do
```

Comment:
Unfortunately this means we are doing the work at compilation time. Which would break cases like this:

```elixir
mod = Mod
forward "/foo", mod
```

We want to do this at runtime. So probably in one of those two places here: https://github.com/phoenixframework/phoenix/blob/4d7198c370818ea3eca2b33b9a4559aa60178e6f/lib/phoenix/router.ex#L779-L780